src/main/kotlin/cacophony/semantic/analysis/VariablesMap.kt

```diff
@@ -0,0 +1,17 @@
+package cacophony.semantic.analysis
+
+import cacophony.controlflow.Variable
+import cacophony.semantic.names.ResolvedVariables
+import cacophony.semantic.syntaxtree.AST
+import cacophony.semantic.syntaxtree.Assignable
+import cacophony.semantic.syntaxtree.Definition
+import cacophony.semantic.types.TypeCheckingResult
+
+data class VariablesMap(
+    val lvalues: Map<Assignable, Variable>,
+    val definitions: Map<Definition.VariableDeclaration, Variable>,
```

Comment:
Are not ArgumentDefinition (or declaration, idk anymore) another option here?

Reply:
Right, will change it to just Definition

Reply:
Another day, another place where a FunctionDefinition should not be a Definition